### PR TITLE
Added systemd service for the rbmc data sync

### PR DESCRIPTION
**Pull request by RameshIyyar**
Please review commit 2. The review for commit 1 is being handled in #1. This is a subsequent PR.

### CODE REVIEW

**Review by harsh-agarwal1 — APPROVED** [review]

File: src/phosphor-rbmc-data-sync-mgr_main.cpp [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+            [](sdbusplus::async::context& ctx) -> sdbusplus::async::task<> {
+            ctx.request_name(BMCDataSync::interface);
+            co_return;
+        }(ctx));
+
+        ctx.run();
+    }
+    catch (std::exception& e)
+    {
+        lg2::error("Exception: {EXCEPTION}", "EXCEPTION", e);
```
Comment (spinler) [inline src/phosphor-rbmc-data-sync-mgr_main.cpp:31]: this means you'll never get a core dump with a backtrace out of this app.  I think that could be useful though to find the source of the exception?

File: src/meson.build [hunk 2]
```diff
@@ -0,0 +1,36 @@
+# SPDX-License-Identifier: Apache-2.0
+
+phosphor_dbus_interfaces = dependency('phosphor-dbus-interfaces')
+phosphor_logging = dependency('phosphor-logging')
+sdbusplus = dependency('sdbusplus')
+
+rbmc_data_sync_sources = [
+    'phosphor-rbmc-data-sync-mgr_main.cpp'
```
Comment (spinler) [inline src/meson.build:8]: That's a pretty long filename.  You prefer that over something like sync_mgr_main.cpp?

File: src/phosphor-rbmc-data-sync-mgr_main.cpp (hunk 1 above)
Comment (spinler) [reply src/phosphor-rbmc-data-sync-mgr_main.cpp:31]: added this commint to PR 1 instead.

File: src/meson.build (hunk 2 above)
Comment (spinler) [reply src/meson.build:8]: added to PR 1

File: src/phosphor-rbmc-data-sync-mgr_main.cpp [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+        sdbusplus::async::context ctx;
+        sdbusplus::server::manager_t objManager{ctx,
+                                                BMCDataSync::namespace_path};
+
+        // clang-tidy currently mangles this into something unreadable
+        // NOLINTNEXTLINE
+        ctx.spawn(
+            [](sdbusplus::async::context& ctx) -> sdbusplus::async::task<> {
+            ctx.request_name(BMCDataSync::interface);
```
[the remark this answers is not in the document]
Comment (spinler) [reply src/phosphor-rbmc-data-sync-mgr_main.cpp:23]: added to PR 1

File: service_files/xyz.openbmc_project.DataSync.BMCData.service [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+After=xyz.openbmc_project.State.BMC.Redundancy.service
+
+[Service]
+ExecStart=/usr/libexec/phosphor-data-sync/phosphor-rbmc-data-sync-mgr
+Restart=always
+Type=dbus
+BusName=xyz.openbmc_project.DataSync.BMCData
+
+[Install]
+WantedBy=multi-user.target
```
Comment (spinler) [inline service_files/xyz.openbmc_project.DataSync.BMCData.service:13]: I think we discussed you'll remove the multi-user.target dep for now?

**Review by aryakpadman — APPROVED** [review]

File: service_files/xyz.openbmc_project.DataSync.BMCData.service (hunk 4 above)
Comment (RameshIyyar) [reply service_files/xyz.openbmc_project.DataSync.BMCData.service:13]: Yes, for now - Just link with active and passive targets

Comment (RameshIyyar) [reply service_files/xyz.openbmc_project.DataSync.BMCData.service:13]: Done

File: src/rbmc_data_sync_main.cpp [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+{
+    using BMCDataSync =
+        sdbusplus::common::xyz::openbmc_project::data_sync::BMCData;
+
+    sdbusplus::async::context ctx;
+    sdbusplus::server::manager_t objManager{ctx, BMCDataSync::namespace_path};
+
+    // clang-tidy currently mangles this into something unreadable
+    // NOLINTNEXTLINE
+    ctx.spawn([](sdbusplus::async::context& ctx) -> sdbusplus::async::task<> {
```
Comment (abhilashraju) [inline src/rbmc_data_sync_main.cpp:18]: can't we call it synchronously? before anything else this should happen right?

Comment (RameshIyyar) [reply src/rbmc_data_sync_main.cpp:18]: Please refer to [this comment](https://github.com/ibm-openbmc/phosphor-data-sync/pull/1#discussion_r1877268267). This PR is specifically intended for reviewing commit 2.

**Review by spinler — APPROVED** [review]

**Pull request merged**